app/lib/features/tasks/pages/task_item_detail_page.dart

```diff
@@ -152,6 +166,54 @@ class TaskItemDetailPage extends ConsumerWidget {
     );
   }
 
+  void showEditDescriptionSheet(
+    BuildContext context,
+    WidgetRef ref,
+    Task task,
+  ) {
+    showEditHtmlDescriptionBottomSheet(
+      context: context,
+      descriptionHtmlValue: task.description()?.formattedBody(),
+      descriptionMarkdownValue: task.description()?.body(),
+      onSave: (newDescription) {
+        final htmlBodyDescription = newDescription.intoHtml();
+        final plainDescription = newDescription.intoMarkdown();
+        _saveDescription(
+          context,
+          ref,
+          task,
+          htmlBodyDescription,
+          plainDescription,
+        );
+      },
+    );
+  }
+
+  Future<void> _saveDescription(
+    BuildContext context,
+    WidgetRef ref,
+    Task task,
+    String htmlBodyDescription,
+    String plainDescription,
+  ) async {
+    EasyLoading.show(status: L10n.of(context).updatingDescription);
+    try {
+      final updater = task.updateBuilder();
+      updater.descriptionHtml(plainDescription, htmlBodyDescription);
+      await updater.send();
+      EasyLoading.dismiss();
+      if (context.mounted) context.pop();
+    } catch (e, st) {
+      _log.severe('Failed to update event description', e, st);
+      EasyLoading.dismiss();
+      if (!context.mounted) return;
+      EasyLoading.showError(
+        L10n.of(context).errorUpdatingDescription(e),
+        duration: const Duration(seconds: 3),
+      );
+    }
```

Comment:
doing this pattern so often, it makes you wonder if we want to generalize this ... (but maybe we want to do that later)

Reply:
Yes true, I was also thinking of the same during this code writing. 
Planning to first later of generalisation and then do this type of second layer of generalisation. :)